tests/test_run_config_generator.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.perf_analyzer.perf_config import PerfAnalyzerConfig
+from model_analyzer.config.generate.run_config_generator import RunConfigGenerator
+from model_analyzer.config.input.config_command_profile \
+     import ConfigCommandProfile
+from model_analyzer.cli.cli import CLI
+from .common import test_result_collector as trc
+from .common.test_utils import convert_to_bytes
+from .mocks.mock_config import MockConfig
+from .mocks.mock_model_config import MockModelConfig
+from .mocks.mock_os import MockOSMethods
+from unittest.mock import MagicMock
+
+from model_analyzer.config.generate.generator_utils import GeneratorUtils as utils
+
+from model_analyzer.config.input.config_defaults import \
+    DEFAULT_BATCH_SIZES, DEFAULT_TRITON_LAUNCH_MODE, \
+    DEFAULT_CLIENT_PROTOCOL, DEFAULT_TRITON_INSTALL_PATH, DEFAULT_OUTPUT_MODEL_REPOSITORY, \
+    DEFAULT_TRITON_INSTALL_PATH, DEFAULT_OUTPUT_MODEL_REPOSITORY, \
+    DEFAULT_TRITON_HTTP_ENDPOINT, DEFAULT_TRITON_GRPC_ENDPOINT, DEFAULT_MEASUREMENT_MODE, \
+    DEFAULT_RUN_CONFIG_MAX_CONCURRENCY, DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT
+
+
+class TestRunConfigGenerator(trc.TestResultCollector):
+
+    def test_default_config_single_model(self):
+        """
+        Test Default Single Model:  
+            - No CLI options specified
+        
+       
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        protobuf = """
```

Comment:
looks like every test has the same protobuf? Any reason not to put it in init or _run_and_test_run_config_generator so they all can use it?